float32 length
---
bool set_ok